### DISCUSSION TITLE
RT tolerance not working in MZmine 3 and 4

### DISCUSSION THREAD

**Discussion by Jihyunchun71** [mzmine/mzmine#1929] [Q&A] [answered]
I am using in-house library (.msp format) with RT information in mzmine 3 and 4.
However, it seems like RT tolerance is totally not working in annotation step.
Should I follow the specific format of RT information in the library file?
My current library format looks like below, please let me know if something should be changed.

e.g.)

NAME: Compound A
RETENTIONTIME: 1.18
FORMULA: CxxHxxOxx
INCHIKEY: xxxx
SMILES: xxxxx
PRECURSORMZ: 378.47578
COLLISIONENERGY: 40.0
PRECURSORTYPE: [M+H]+
IONMODE: Positive
Num Peaks: 21
69.5142 17306.67
187.0542 63546.46
196.0664 3349.22
257.0699 7155.468
378.47578 4845.548

**Comment by ansgarkorf** [comment, marked as answer]
Hi,
please have a look at the getNistMspID() method. Can you try to change "RETENTIONTIME" in your library entry to "RT"?

/**
   * @return The NIST MSP format key or an empty String
   */
  public String getNistMspID() {
    return switch (this) {
      case SCAN_NUMBER -> "scan_number";
      case MERGED_SPEC_TYPE -> "merge_type";
      case ENTRY_ID -> "DB#";
      case COLLISION_ENERGY -> "Collision_energy";
      case COMMENT -> "Comments";
      case EXACT_MASS -> "ExactMass";
      case FRAGMENTATION_METHOD -> "Method";
      case ISOLATION_WINDOW -> "Isolation_window";
      case FORMULA -> "Formula";
      case INCHIKEY -> "InChIKey";
      case INSTRUMENT -> "Instrument";
      case INSTRUMENT_TYPE -> "Instrument_type";
      case ION_TYPE -> "Precursor_type";
      case POLARITY -> "Ion_mode"; // P / N
      case ION_SOURCE -> "";
      case IMS_TYPE -> "ims_type";
      case PRECURSOR_MZ -> "PrecursorMZ";
      case NAME -> "Name";
      case SPLASH -> "Splash";
      case RT -> "RT";
      case MS_LEVEL -> "Spectrum_type";
      case NUM_PEAKS -> "Num Peaks";
      case CCS -> "CCS";
      case SMILES -> "SMILES";
      case INCHI -> "INCHI";
      case ACQUISITION, GNPS_ID, MONA_ID, CHEMSPIDER, RESOLUTION, SYNONYMS, MOLWEIGHT, PUBCHEM, PUBMED, PRINCIPAL_INVESTIGATOR, CHARGE, CAS, SOFTWARE, DATA_COLLECTOR ->
          this.name().toLowerCase();
      case PEPTIDE_SEQ -> "peptide_sequence";
      case MSN_COLLISION_ENERGIES -> "MSn_collision_energies";
      case MSN_PRECURSOR_MZS -> "MSn_precursor_mzs";
      case MSN_FRAGMENTATION_METHODS -> "MSn_fragmentation_methods";
      case MSN_ISOLATION_WINDOWS -> "MSn_isolation_windows";
      case USI -> "usi";
      case DESCRIPTION -> "description";
      case QUALITY -> "quality";
      case DATASET_ID -> "dataset_id";
      case QUALITY_CHIMERIC -> "quality_chimeric";
      case QUALITY_PRECURSOR_PURITY -> "precursor_purity";
      case QUALITY_EXPLAINED_INTENSITY -> "quality_explained_intensity";
      case QUALITY_EXPLAINED_SIGNALS -> "quality_explained_signals";
      case OTHER_MATCHED_COMPOUNDS_N -> "other_matched_compounds";
      case OTHER_MATCHED_COMPOUNDS_NAMES -> "other_matched_compounds_names";
      case FEATURE_ID -> "feature_id";
      case FILENAME -> "file_name";
      case ONLINE_REACTIVITY -> "online_reactivity";
      case FEATURE_MS1_HEIGHT -> "feature_ms1_height";
      case FEATURE_MS1_REL_HEIGHT -> "feature_ms1_relative_height";
      case SIRIUS_MERGED_SCANS -> "";
      case SIRIUS_MERGED_STATS -> "";
      case UNSPECIFIED -> "";
    };
  }
